pkg/tsdb/intervalv2/intervalv2.go

```diff
@@ -53,16 +53,21 @@ func (i *Interval) Milliseconds() int64 {
 	return i.Value.Nanoseconds() / int64(time.Millisecond)
 }
 
-func (ic *intervalCalculator) Calculate(timerange backend.TimeRange, minInterval time.Duration) Interval {
+func (ic *intervalCalculator) Calculate(timerange backend.TimeRange, minInterval time.Duration, maxDataPoints int64) Interval {
 	to := timerange.To.UnixNano()
 	from := timerange.From.UnixNano()
-	calculatedIntrvl := time.Duration((to - from) / defaultRes)
+	resolution := maxDataPoints
+	if resolution == 0 {
```

Comment:
in theory this should never happen anymore, all paths that lead here should calculate maxDataPoints correctly.

but i understand why the check is here for now... i wonder if we should put here some logging perhaps or something that notifies us that we reached a place that should not happen and fixing is needed 😄

Reply:
@gabor we can log this, but for us to make this log helpful, we would need to know what query resulted in `maxDataPoinits == 0`. Which would. end up with adding more arguments/passiing plog as in `customQueryParametersMiddleware(plog)`. Or we could check it in each data source beforehand?  Do you have any thoughts on which one should be added?

Reply:
@gabor I am going to merge this so I can move on with Prometheus backend refactor, but I added this as a task in https://github.com/grafana/grafana/issues/37784. We can have a chat later about how to notify us about `maxDataPoinits == 0`.